core/src/main/java/org/apache/brooklyn/core/mgmt/internal/LocalEntityManager.java

```diff
@@ -742,7 +742,7 @@ private boolean unmanageNonRecursive(Entity e) {
             if (e instanceof Group) {
                 Collection<Entity> members = ((Group)e).getMembers();
                 for (Entity member : members) {
-                    if (!Entities.isNoLongerManaged(member)) member.removeGroup((Group)e);
+                    if (!Entities.isNoLongerManaged(member)) ((EntityInternal)member).groups().remove((Group)e);
```

Comment:
`removeGroup` is deprecated, but it's just defined as 
```java
   public void removeGroup(Group group) {
        groups().remove(group);
    }
```
so I don't see how this change is changing anything really? Is it not just doing the same thing? The comments on removeGroup are slightly ambiguous but seem to suggest that the preferred thing to be doing here is calling `Group#removeMember`, which will update the group size sensors and so on.  Same thought applies to the changes to `add` below.